nexus/src/sagas.rs

```diff
@@ -1294,81 +1336,69 @@ async fn sdc_regions_ensure(
+            }));
         }
-
-        // If the size of the image is greater than the size of the disk,
-        // return an error.
-        if global_image.size.to_bytes() > params.create_params.size.to_bytes() {
-            return Err(ActionError::action_failed(Error::invalid_request(
-                &format!(
-                    "image size {} is greater than disk size {}",
-                    global_image.size.to_bytes(),
-                    params.create_params.size.to_bytes()
-                ),
-            )));
+        params::DiskSource::Image { image_id: _ } => {
+            // Until we implement project images, do not allow disks to be
+            // created from a project image.
+            return Err(ActionError::action_failed(Error::InvalidValue {
+                label: String::from("image"),
+                message: String::from("project image are not yet supported"),
+            }));
         }
+        params::DiskSource::GlobalImage { image_id } => {
+            warn!(log, "grabbing image {}", image_id);
 
-        warn!(
-            log,
-            "grabbing global image {} volume {}",
-            global_image.id(),
-            global_image.volume_id
-        );
-
-        let volume = osagactx
-            .datastore()
-            .volume_get(global_image.volume_id)
-            .await
-            .map_err(|e| {
-                ActionError::action_failed(Error::internal_error(
-                    &e.to_string(),
-                ))
-            })?;
-
-        warn!(
-            log,
-            "grabbed volume {}, with data {}",
-            volume.id(),
-            volume.data()
-        );
-
-        Some(Box::new(serde_json::from_str(volume.data()).map_err(|e| {
-            ActionError::action_failed(Error::internal_error(&e.to_string()))
-        })?))
-    } else {
-        None
+            let (.., global_image) =
+                LookupPath::new(&opctx, &osagactx.datastore())
+                    .global_image_id(*image_id)
+                    .fetch()
+                    .await
+                    .map_err(ActionError::action_failed)?;
+
+            debug!(log, "retrieved global image {}", global_image.id());
+
+            debug!(
+                log,
+                "grabbing global image {} volume {}",
+                global_image.id(),
+                global_image.volume_id
+            );
+
+            let volume = osagactx
+                .datastore()
+                .volume_get(global_image.volume_id)
+                .await
+                .map_err(ActionError::action_failed)?;
+
+            debug!(
+                log,
+                "grabbed volume {}, with data {}",
+                volume.id(),
+                volume.data()
+            );
+
+            Some(Box::new(serde_json::from_str(volume.data()).map_err(
+                |e| {
+                    ActionError::action_failed(Error::internal_error(
+                        &e.to_string(),
```

Comment:
maybe want a little more context in this message? ("failed to deserialize volume data"?)

Reply:
:+1: added in 47523593